src/events/index.spec.js

```diff
@@ -0,0 +1,341 @@
+    context('the event ID is provided', function() {
+      let eventFromServerAfterFormat;
+      let eventFromServerBeforeFormat;
+      let expectedEventId;
+      let formatEventFromServer;
+      let promise;
+      let request;
+
+      beforeEach(function() {
+        expectedEventId = faker.random.uuid();
+        eventFromServerAfterFormat = fixture.build(
+          'event',
+          { id: expectedEventId },
+          {
+            fromServer: false
+          }
+        );
+        eventFromServerBeforeFormat = fixture.build(
+          'event',
+          { id: expectedEventId },
+          {
+            fromServer: true
+          }
+        );
+
+        formatEventFromServer = this.sandbox
+          .stub(eventsUtils, 'formatEventFromServer')
+          .returns(eventFromServerAfterFormat);
+
+        request = {
+          ...baseRequest,
+          get: this.sandbox.stub().resolves(eventFromServerBeforeFormat)
+        };
+
+        const events = new Events(baseSdk, request);
+        events._baseUrl = expectedHost;
+
+        promise = events.get(expectedEventId);
+      });
+
+      it('gets the event from the server', function() {
+        expect(request.get).to.be.calledWith(
+          `${expectedHost}/events/${expectedEventId}`
+        );
+      });
+
+      it('formats the event object', function() {
+        return promise.then(() => {
+          expect(formatEventFromServer).to.be.calledWith(
+            eventFromServerBeforeFormat
+          );
+        });
+      });
+
+      it('returns the requested event', function() {
+        return expect(promise).to.be.fulfilled.and.to.eventually.deep.equal(
+          eventFromServerAfterFormat
+        );
+      });
+    });
+
+    context('the event Id is not provided', function() {
+      it('throws an error', function() {
+        const events = new Events(baseSdk, baseRequest);
+        const promise = events.get();
+
+        return expect(promise).to.be.rejectedWith(
+          'An event ID is required for getting information about an event'
+        );
+      });
+    });
+  });
+
+  describe('update', function() {
+    context('when all required information is available', function() {
+      let eventFromServerBeforeFormat;
+      let eventToServerAfterFormat;
+      let eventToServerBeforeFormat;
+      let formatEventToServer;
+      let request;
+      let promise;
+
+      beforeEach(function() {
+        eventFromServerBeforeFormat = fixture.build('event', null, {
+          fromServer: true
+        });
+        eventToServerAfterFormat = fixture.build('event', null, {
+          fromServer: true
+        });
+        eventToServerBeforeFormat = fixture.build('event');
+
+        formatEventToServer = this.sandbox
+          .stub(eventsUtils, 'formatEventToServer')
+          .returns(eventToServerAfterFormat);
+
+        request = {
+          ...baseRequest,
+          put: this.sandbox.stub().resolves(eventFromServerBeforeFormat)
```

Comment:
I'd resolve with nothing here, since only a 204 is returned from the API

Reply:
Ah yeah, good call!